content/blog/coredns-in-kubernetes-via-kubeadm.md

```diff
@@ -0,0 +1,267 @@
+[apiclient] Found 1 Pods for label selector component=kube-scheduler
+[upgrade/staticpods] Component "kube-scheduler" upgraded successfully!
+[uploadconfig] Storing the configuration used in ConfigMap "kubeadm-config" in the "kube-system" Namespace
+[bootstraptoken] Configured RBAC rules to allow Node Bootstrap tokens to post CSRs in order for nodes to get long term certificate credentials
+[bootstraptoken] Configured RBAC rules to allow the csrapprover controller automatically approve CSRs from a Node Bootstrap Token
+[bootstraptoken] Configured RBAC rules to allow certificate rotation for all node client certificates in the cluster
+[addons] Applied essential addon: CoreDNS
+[addons] Applied essential addon: kube-proxy
+
+[upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.10.0-alpha.1". Enjoy!
+
+[upgrade/kubelet] Now that your control plane is upgraded, please proceed with upgrading your kubelets in turn.
+
+~~~
+
+To verify CoreDNS is running, we can check the pod status, deployment and the configmap in the node.
+
+Check `pod` status:
+~~~ text
+# kubectl -n kube-system get pods -o wide
+NAME                               READY     STATUS    RESTARTS   AGE       IP              NODE
+coredns-546545bc84-ttsh4           1/1       Running   0          5h        10.32.0.61      sandeep2
+etcd-sandeep2                      1/1       Running   0          5h        147.75.107.43   sandeep2
+kube-apiserver-sandeep2            1/1       Running   0          4h        147.75.107.43   sandeep2
+kube-controller-manager-sandeep2   1/1       Running   0          4h        147.75.107.43   sandeep2
+kube-proxy-fkxmg                   1/1       Running   0          4h        147.75.107.43   sandeep2
+kube-scheduler-sandeep2            1/1       Running   4          5h        147.75.107.43   sandeep2
+weave-net-jhjtc                    2/2       Running   0          5h        147.75.107.43   sandeep2
+~~~
+
+Check `Deployment`:
+~~~ text
+# kubectl -n kube-system get deployments
+NAME      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+coredns   1            1          1           1        4h
+~~~
+
+Check the `Configmap`:
+~~~ text
+# kubectl -n kube-system get configmap coredns -oyaml
+apiVersion: v1
+data:
+  Corefile: |
+    .:53 {
+        errors
+        health
+        kubernetes cluster.local 10.96.0.0/12 {
+           pods insecure
+           upstream /etc/resolv.conf
+        }
+        prometheus :9153
+        proxy . /etc/resolv.conf
+        cache 30
+    }
+kind: ConfigMap
+metadata:
+  creationTimestamp: 2017-12-21T12:55:15Z
+  name: coredns
+  namespace: kube-system
+  resourceVersion: "161"
+  selfLink: /api/v1/namespaces/kube-system/configmaps/coredns
+  uid: 30bf0882-e64e-11e7-baf6-0cc47a8055d6
+~~~
+
+>It is important to note that when switching from kube-dns to CoreDNS, the configurations that come with kube-dns (stubzones, federations...) will no longer exist and will switch to a default configuration in CoreDNS.
```

Comment:
I guess we should say also that CoreDNS is complely compatible with kube-dns. We can point to existing document.